imports/wasi_snapshot_preview1/wasi_stdlib_test.go

```diff
@@ -217,18 +220,83 @@ func compileAndRun(t *testing.T, config wazero.ModuleConfig, bin []byte) (consol
 }
 
 func Test_Poll(t *testing.T) {
-	moduleConfig := wazero.NewModuleConfig().WithArgs("wasi", "poll")
-	console := compileAndRun(t, moduleConfig, wasmZigCc)
-	// The "real" expected behavior is to return "NOINPUT",
-	// however the poll API is currently relying on stat'ing the file
-	// descriptor for stdin which makes the behavior platform-specific
-	// **during tests** and unfortunately hard to mock.
-	// For now, we just make sure the result is consistent.
-	if stat, err := os.Stdin.Stat(); err != nil {
-		if stat.Mode()&fs.ModeCharDevice != 0 {
-			require.Equal(t, "NOINPUT\n", console)
-			return
-		}
+	// The following test cases replace Stdin with a custom reader.
+	// For more precise coverage, see poll_test.go.
+
+	tests := []struct {
+		name            string
+		args            []string
+		stdin           io.Reader
+		expectedOutput  string
+		expectedTimeout time.Duration
+	}{
+		{
+			name: "custom reader, data ready, not tty",
+			args: []string{"wasi", "poll"},
+			stdin: internalsys.NewStdioFileReader(
+				bufio.NewReader(strings.NewReader("test")), // input ready
+				stdinFileInfo(fs.ModeDevice|fs.ModeCharDevice|0o640),
+				internalsys.PollerAlwaysReady),
+			expectedOutput:  "STDIN",
+			expectedTimeout: 0 * time.Millisecond,
+		},
+		{
+			name: "custom reader, data ready, not tty, .5sec",
```

Comment:
TBH I am not very happy about these time-sensitive tests as they obviously tend to become flaky, as I am noticing right now. Suggestions welcome

Reply:
I think you need a larger timeout. for example, we have a comment like this in Test_Nanosleep

```
	// In CI, Nanosleep(50ms) returned after 197ms.
	// As we can't control the platform clock, we have to be lenient
	ns := int64(50 * time.Millisecond)
```

Reply:
yeah I increased them to 500 millis, hope that's good enough